File: BlackJack.ViewModels/GameViews/GetAllStepsByPlayerIdAndGameIdGameView.cs
using BlackJack.ViewModels.EnumViews;
using System;
using System.Collections.Generic;

namespace BlackJack.ViewModels.GameViews
{
    public class GetAllStepsByPlayerIdAndGameIdGameView
    {
        public List<PlayerStepGetAllStepsByPlayerIdAndGameIdGameViewItem> PlayerSteps { get; set; }

        public GetAllStepsByPlayerIdAndGameIdGameView()
        {
            PlayerSteps = new List<PlayerStepGetAllStepsByPlayerIdAndGameIdGameViewItem>();
        }
    }

    public class PlayerStepGetAllStepsByPlayerIdAndGameIdGameViewItem
    {
        public Guid Id { get; set; }
        public SuiteTypeEnumView Suite { get; set; }
        public RankTypeEnumView Rank { get; set; }
        public PlayerGetAllStepsByPlayerIdAndGameIdGameView Player { get; set; }
        public GameGetAllStepsByPlayerIdAndGameIdGameView Game { get; set; }
    }

    public class PlayerGetAllStepsByPlayerIdAndGameIdGameView
    {
        public string PlayerId { get; set; }
        public string UserName { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }

    public class GameGetAllStepsByPlayerIdAndGameIdGameView
    {
        public Guid GameId { get; set; }
        public GameStateTypeEnumView GameState { get; set; }
    }
}

File: BlackJack.WEB/ClientApp/src/app/shared/entities/game.views/get-all-steps-by-player-id-and-game-id.game.view.ts
import { SuiteType } from '../../enums/suite-type';
import { RankType } from '../../enums/rank-type';
import { GameStateType } from '../../enums/game-state-type';


export class GetAllStepsByPlayerIdAndGameIdGameView {
  playerSteps?: Array<PlayerStepGetAllStepsByPlayerIdAndGameIdGameViewItem>;

  constructor() {
    this.playerSteps = new Array<PlayerStepGetAllStepsByPlayerIdAndGameIdGameViewItem>();
  }
}

export class PlayerStepGetAllStepsByPlayerIdAndGameIdGameViewItem {
  id?: string;
  suite?: SuiteType;
  rank?: RankType;
  player?: PlayerGetAllStepsByPlayerIdAndGameIdGameView;
  game?: GameGetAllStepsByPlayerIdAndGameIdGameView;

  constructor() {
    this.player = new PlayerGetAllStepsByPlayerIdAndGameIdGameView();
    this.game = new GameGetAllStepsByPlayerIdAndGameIdGameView();
  }
}

export class PlayerGetAllStepsByPlayerIdAndGameIdGameView {
  id?: string;
  balance?: number;
  bet?: number;
}

export class GameGetAllStepsByPlayerIdAndGameIdGameView {
  gameId?: string;
  gameState?: GameStateType;
}
